docs/extend/extend-apiml/api-mediation-oidc-authentication.md

```diff
@@ -0,0 +1,87 @@
+After successful login the OIDC provider grants the client an Access Token (along with Identity Token, eventually also a Refresh Token if requested and appropriate).
+The Access Token issued by a trusted OIDC provider is now accepted by APIML.
+is the foundation of API ML multi-domain Identity Federation.
+The JWT Access Token obtained from the OIDC/IDP can be then used to authenticate the user (and authorize the client application) and allow access to mainframe resources through the API ML GW.
+
+## Authentication Flow
+The following diagram describes the interactions between the participants of the OIDC based API ML authentication process.
+
+<img src={require("../../images/api-mediation/apiml-oidc-auth-seq.png").default} alt="APIML OIDC Workflow" width="700"/>
+
+* When the User wants to access mainframe resources or services using the client application without a valid authentication / access token, 
+the client redirects the User agent to the login end-point of the distributed OIDC provider. 
+* The user is asked to provide their valid credentials (authentication factors).
+* After successful validation of all authentication factors, the OIDC provider grants the Client an access token.
+* The client then can request from API ML GW the needed mainframe resources presenting the access token in the request. 
+* The GW validates the access token at the OIDC provider's /introspection end-point and in case of success, caches the outcome for a short time. 
+* In subsequent calls with the same token, the GW reuses the cached validation outcome. This helps to save round trips to the OIDC /introspection end-point 
+for a short interval of time, when the client needs to access multiple resources in a row to complete a unit of work. The caching interval is configurable with a sensible default. 
+* The API ML GW fetches  the distributed user identity from the distributed access token and maps it to the user mainframe identity using SAF. 
+* The mainframe user identity is used to create the mainframe user credentials (Zowe or SAF JWT or pass-ticket) expected by the target mainframe service.
+
+## API ML OIDC Configuration
+
+### Prerequisites
+
+- The users who need access to mainframe resources using OIDC authentication must have a mainframe identity managed by SAF/ESM
+- The users need to have distributed identity managed by the OIDC provider  
+- SAF/ESM is configured with mapping between the mainframe and distributed user identities
+  
+### OIDC provider configuration
+
+- Client Application configuration in the OIDC provider. 
+
+  Depending on the OIDC provider and client application capabilities the configuration of the OIDC provider would vary.
+For example Web Applications with a secure server side component can use `code grant authorization flow` and can be granted Refresh Token, whereas a Single Page Application running entirely in the User Agent (browser) is more limited regarding its security capabilities.  
```

Comment:
Should we define "code grant authorization flow", or provide a link to where this term is described?

Reply:
There is a link to the protocols - OIDC and OAuth2 at the beginning of the document. Also, I'd expect that people interested in this functionality should know what it means.